services/venice-server/src/main/java/com/linkedin/venice/listener/StatsHandler.java

```diff
@@ -294,8 +285,11 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
 
   private void recordBasicMetrics() {
     if (storeName != null) {
+      ServerHttpRequestStats serverHttpRequestStats = currentStats.getStoreStats(storeName);
+      ServerHttpRequestStats totalStats = currentStats.getTotalStats();
       if (databaseLookupLatency >= 0) {
-        currentStats.recordDatabaseLookupLatency(storeName, databaseLookupLatency, isAssembledMultiChunkLargeValue());
+        totalStats.recordDatabaseLookupLatency(databaseLookupLatency, isAssembledMultiChunkLargeValue());
+        serverHttpRequestStats.recordDatabaseLookupLatency(databaseLookupLatency, isAssembledMultiChunkLargeValue());
```

Comment:
The refactoring in this function should successfully eliminate many map lookups, which is great, but I think the resulting code seems brittle.

Ideally, the store-level stat would take care of also updating the total stat. This can be achieved by making each function of the store-level stat also invoke the total stat, which is fine from an encapsulation standpoint, but also fairly repetitive. Or it can also be done by registering the total stat as a parent to the store-level stat at the time of creating the sensor. See the host level ingestion stat for an example of this last approach.

Reply:
I was trying to avoid that as it seemed needed lot more refactoring + chances of messing up names and type of stats associated with. I bit the bulltet and followed the pattern in host level ingestion stat.

Reply:
Thanks for biting the bullet :D ... it looks great, IMHO.